language/golang/configure-ci-cd.md

```diff
@@ -8,331 +8,7 @@ redirect_from:
-7a1dde643d3d: Pull complete
-f815a8b426ad: Pull complete
-7a6b1ee48c34: Pull complete
-ca1a2b73aa81: Pull complete
-Digest: sha256:81bedd562952757322a07a26634b01e3916db375cc695843124f79641e433029
-Status: Downloaded newer image for olliefr/docker-gs-ping:latest
-docker.io/olliefr/docker-gs-ping:latest
-```
-
-We've just build a simple (maybe even naive) CI/CD workflow. There are many ways in which it can be improved. We'll look at some of these ways in the next section.
-
-## Optimizing the workflow
-
-Next, let’s look at how we can optimize the GitHub Actions workflow through build cache. This has two main advantages:
-
-1. Build cache reduces the build time as it will not have to re-download all of the images, and
-2. It also reduces the number of pulls we complete against Docker Hub. We need to make use of GitHub cache to make use of this.
-
-Let us set up a Builder with a build cache. First, we need to set up the builder, and then to configure the cache. In this example, let us add the path and keys to store this under using GitHub cache for this.
-
-{% raw %}
-```yaml
-      - name: Set up Docker Buildx
-        id:   buildx
-        uses: docker/setup-buildx-action@v1
-
-      - name: Cache Docker layers
-        uses: actions/cache@v2
-        with:
-          path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-buildx-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-buildx-
-```
-{% endraw %}
-
-And lastly, after adding the builder and build cache snippets to the top of the Actions file, we need to add some extra attributes to the build and push step. This involves:
-
-* Setting up the builder to use the output of the buildx step, and then
-* Using the cache we set up earlier for it to store to and to retrieve
-
-{% raw %}
-```yaml
-      - name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
-        with:
-          builder: ${{ steps.buildx.outputs.name }}
-          push: false
-          load: true
-          tags: ${{ github.event.repository.name }}:latest, ${{ github.repository }}:latest
-          cache-from: type=local,src=/tmp/.buildx-cache
-          cache-to: type=local,dest=/tmp/.buildx-cache
-```
-{% endraw %}
-
-Now, run the CI workflow again and verify that it uses the build cache by checking the workflow log.
-
-## Wrap up
-
-GitHub Actions are an immensely powerful way to automate your CI and CD pipelines, and [there is a lot of documentation](https://docs.github.com/en/actions) to aid you in that mission. It is not, however, the _only_ way to integrate containers into your workflow. The aim of this section was to show you some of the basic things that are possible. There is a multitude of CI and CD tools on the market and you are very welcome to investigate what integration with the container ecosystem they provide. Well defined, automated CI pipelines can save you and your team a lot of effort.
+{% include gha-tutorial.md %}
```

Comment:
@jedevc @crazy-max @thaJeztah I have replaced all of the CI/CD guides in `/language` with an include, reusing the same tutorial everywhere. Before, all guides contained basically a copy/paste version of the guide, _except for golang_ where the CI/CD guide was purpose-built.

Do you think it would be a problem to replace the 🦄 unique 🦄  golang CI guide with a generic one?